lib/capsize/application.rb

```diff
@@ -0,0 +1,31 @@
+module Capsize
+  module Application
+    def self.load_tasks
+      Application.load_tasks_in_isolation do
+        require 'capistrano/all'
+        cap = Capistrano::Application.new
+        Rake.application = cap
+        cap.init
+        cap.load_rakefile
+        cap.tasks
+      end
+    end
+
+    def self.load_tasks_in_isolation
+      read, write = IO.pipe
+
+      pid = fork do
+        read.close
+        result = yield
+        write.puts result
```

Comment:
I would pass `write` into the `yield` call so that you don't have the intermediate step of collecting it all into a local variable: just let the calling code write to the result directly.